pkg/config/config.go

```diff
@@ -129,3 +129,26 @@ func (c *Config) LoadIncludes(ctx context.Context, client scm.Client) error {
 
 	return nil
 }
+
+// Merge merges the other config into the current config
+func (c *Config) Merge(other *Config) *Config {
```

Comment:
[suggestion] This function seems to both:

1. Return the merged configs; and also,
2. Mutate the original `c *Config`

Why both? Is that just for convenience?

I would have probably expected this to only do one or the other.

For example, if we wanted to avoid mutating the original, I would have expected this to create and return a new `Config` struct, any maybe avoid using pointers in the func signature.  Otherwise, if the plan was to mutate, I would have expected there to be no return type.

I personally prefer erring on the side of immutability where possible, but I'll defer to you here.  Feel free to revise this using either option, or to tell me you prefer the code as-is!

Reply:
i'm fine with mutating the original config since its set early on/ way before any configs are actually used, so i'll remove returning the actual value since it will be mutable. will update